# Generating a Tree Using a Tutorial
### Julien Dhouti

We first import all of the libraries that we need.

In [2]:
library(caret)

We import the data using the `read_csv()` function from the `readr` library. From this, we select only the columns that we may need such as:
* Survived
* Embarked
* Sex
* SibSp
* Parch
* Fare

In [8]:
titanic3 <- read.csv("datasets/train.csv", stringsAsFactors = T)

# view the data that we just imported
str(titanic3)

'data.frame':	891 obs. of  12 variables:
 $ PassengerId: int  1 2 3 4 5 6 7 8 9 10 ...
 $ Survived   : int  0 1 1 1 0 0 0 0 1 1 ...
 $ Pclass     : int  3 1 3 1 3 3 1 3 3 2 ...
 $ Name       : Factor w/ 891 levels "Abbing, Mr. Anthony",..: 109 191 358 277 16 559 520 629 417 581 ...
 $ Sex        : Factor w/ 2 levels "female","male": 2 1 1 1 2 2 2 2 1 1 ...
 $ Age        : num  22 38 26 35 35 NA 54 2 27 14 ...
 $ SibSp      : int  1 1 0 1 0 0 0 3 0 1 ...
 $ Parch      : int  0 0 0 0 0 0 0 1 2 0 ...
 $ Ticket     : Factor w/ 681 levels "110152","110413",..: 524 597 670 50 473 276 86 396 345 133 ...
 $ Fare       : num  7.25 71.28 7.92 53.1 8.05 ...
 $ Cabin      : Factor w/ 148 levels "","A10","A14",..: 1 83 1 57 1 1 131 1 1 1 ...
 $ Embarked   : Factor w/ 4 levels "","C","Q","S": 4 2 4 4 4 3 4 4 4 2 ...


Before we can process the data, we should always make sure to see if there are any missing values.

In [9]:
sum(is.na(titanic3))

[1] 177

By the looks of it, it appears that there are 177 missing values in the dataset. Let's use KNN to fix these values.

In [14]:
# Imputing the missing values
preProcValues <- preProcess(titanic3, method = c("scale", "center", "knnImpute"))

library("RANN")
train_processed <- predict(preProcValues, titanic3)

Now let's see if there are any missing values that remain.

In [15]:
sum(is.na(train_processed))

[1] 0